packages/@sanity/components/src/progress/ProgressCircle.js

```diff
@@ -6,86 +6,77 @@ const radiusFactor = 1.3
 const widthFactor = 1.2
 const heightFactor = 1.2
 
-export default class ProgressCircle extends React.Component {
+export default class ProgressCircle extends React.PureComponent {
   static propTypes = {
-    className: PropTypes.string,
     percent: PropTypes.number,
-    animation: PropTypes.bool,
     text: PropTypes.string,
     style: PropTypes.object,
-    showPercent: PropTypes.bool,
-    radius: PropTypes.number,
-    strokeWidth: PropTypes.number,
-    status: PropTypes.string,
-    completed: PropTypes.bool
+    showPercent: PropTypes.bool
   }
 
   static defaultProps = {
-    radius: 50,
-    percentage: 0,
-    strokeWidth: 10
+    percent: 0,
+    style: {},
+    text: undefined,
```

Comment:
`text` should be removed from defaultProps here as it is redundant.

Reply:
The linter says I should have it. Think it is the samme issue as above.

Reply:
Yeah, I think we should switch that rule off entirely.

Reply:
On second thought, I think the issue is rather that the default type is `undefined` and not an empty string here.

Specifying defaults for non-required props is a good thing, but they should always be of the same type as the prop, so when the prop is an array the default should be `[]`, if string, `''`, etc. This way you can blindly call properties and methods defined on the prop value (e.g. `text.substring(0, 10)` without having to first do a undefined check.

Reply:
Ah. That makes sense